presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -393,6 +394,7 @@ else if (serverConfig.isCoordinator()) {
         jaxrsBinder(binder).bind(TaskExecutorResource.class);
         newExporter(binder).export(TaskExecutorResource.class).withGeneratedName();
         binder.bind(TaskManagementExecutor.class).in(Scopes.SINGLETON);
+        binder.bind(LowMemoryMonitor.class).in(Scopes.SINGLETON);
```

Comment:
Is there a way to only inject this for the worker ? I am not sure if this would also inject it for the coordinator as well ? I know the coordinator can run splits as well, but do we want this throttling there ?

Reply:
Yes we can do that. Moving it to `WorkerModule`.